tar/bsdtar.c

```diff
@@ -876,6 +878,20 @@ main(int argc, char **argv)
 	if (bsdtar->strip_components != 0)
 		only_mode(bsdtar, "--strip-components", "xt");
 
+	/* Attempt to load keyfile. */
+	if (bsdtar->keyfile != NULL) {
+		if (load_keys(bsdtar, bsdtar->keyfile) == 0)
+			bsdtar->have_keys = 1;
+		else {
+			if (bsdtar->option_dryrun)
```

Comment:
How about setting a `bsdtar->keyfile_from_config` option so that a command-line `--keyfile` directive doesn't get the "ignore if unavailable" dry-run handling?